cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,454 @@
+	var containerSCC ContainerSCC
+	containerSCC.HostIPC = NOK
+	if pod.Spec.HostIPC {
+		containerSCC.HostIPC = OK
+	}
+	containerSCC.HostNetwork = NOK
+	if pod.Spec.HostNetwork {
+		containerSCC.HostNetwork = OK
+	}
+	containerSCC.HostPID = NOK
+	if pod.Spec.HostPID {
+		containerSCC.HostPID = OK
+	}
+	containerSCC.SeLinuxContextPresent = NOK
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContextPresent = OK
+	}
+	containerSCC.AllVolumeAllowed, containerSCC.HostDirVolumePluginPresent = AllVolumeAllowed(pod.Spec.Volumes)
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUserPresent = OK
+	} else {
+		containerSCC.RunAsUserPresent = NOK
+	}
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
+		containerSCC.FsGroupPresent = OK
+	} else {
+		containerSCC.FsGroupPresent = NOK
+	}
+	return CheckCategory(pod.Spec.Containers, containerSCC, pod.Name, pod.Namespace)
+}
+
+//nolint:gocyclo,funlen
+func compareCategory(refCategory, containerSCC *ContainerSCC, id CategoryID) bool {
+	result := true
+	tnf.ClaimFilePrintf("Testing if pod belongs to category %s", &id)
+	if refCategory.AllVolumeAllowed == containerSCC.AllVolumeAllowed {
+		tnf.ClaimFilePrintf("AllVolumeAllowed = %s - OK", containerSCC.AllVolumeAllowed)
+	} else {
+		result = false
+		tnf.ClaimFilePrintf("AllVolumeAllowed = %s but expected >=<=%s -  NOK", containerSCC.AllVolumeAllowed, refCategory.AllVolumeAllowed)
+	}
+	if refCategory.RunAsUserPresent == containerSCC.RunAsUserPresent {
+		tnf.ClaimFilePrintf("RunAsUserPresent = %s - OK", containerSCC.RunAsUserPresent)
+	} else {
+		tnf.ClaimFilePrintf("RunAsUserPresent = %s but expected  %s - NOK", containerSCC.RunAsUserPresent, refCategory.RunAsUserPresent)
+		result = false
+	}
+	if refCategory.RunAsNonRoot >= containerSCC.RunAsNonRoot {
+		tnf.ClaimFilePrintf("RunAsNonRoot = %s - OK", containerSCC.RunAsNonRoot)
+	} else {
+		tnf.ClaimFilePrintf("RunAsNonRoot = %s but expected  %s - NOK", containerSCC.RunAsNonRoot, refCategory.RunAsNonRoot)
+		result = false
+	}
+	if refCategory.FsGroupPresent == containerSCC.FsGroupPresent {
+		tnf.ClaimFilePrintf("FsGroupPresent  = %s - OK", containerSCC.FsGroupPresent)
+	} else {
+		tnf.ClaimFilePrintf("FsGroupPresent  = %s but expected  %s - NOK", containerSCC.FsGroupPresent, refCategory.FsGroupPresent)
+		result = false
+	}
+	if refCategory.HaveDropCapabilities == containerSCC.HaveDropCapabilities {
+		tnf.ClaimFilePrintf("DropCapabilities list - OK")
+	} else {
+		tnf.ClaimFilePrintf("HaveDropCapabilities = %s but expected  %s - NOK", containerSCC.HaveDropCapabilities, refCategory.HaveDropCapabilities)
+		tnf.ClaimFilePrintf("its didnt have all the required (MKNOD, SETUID, SETGID, KILL)/(ALL) drop value ")
+		result = false
+	}
+	if refCategory.HostDirVolumePluginPresent == containerSCC.HostDirVolumePluginPresent {
```

Comment:
Add Comment:
// HostDirVolumePluginPresent is true if a hostpath volume is configured, false otherwise. It is a deprecated field and is derived from the volume list currently configured in the container. see https://docs.openshift.com/container-platform/3.11/admin_guide/manage_scc.html#use-the-hostpath-volume-plugin